cpp/src/arrow/acero/hash_aggregate_test.cc

```diff
@@ -592,6 +592,11 @@ void TestSegments(std::unique_ptr<RowSegmenter>& segmenter, const ExecSpan& batc
     ASSERT_EQ(expected_segment, segment);
     offset = segment.offset + segment.length;
   }
+  // Assert next is the last (empty) segment.
+  ASSERT_OK_AND_ASSIGN(auto segment, segmenter->GetNextSegment(batch, offset));
+  ASSERT_TRUE(segment.offset >= batch.length);
```

Comment:
Nit: use richer assertion
```suggestion
  ASSERT_GE(segment.offset, batch.length);
```